File: Apollo.Domain/DTO/User/SocietyUser.cs
using Apollo.Core.Interface;
using Apollo.Domain.Enum;
using System;
using System.Collections.Generic;
using System.Text;

namespace Apollo.Domain.DTO.User
{
    public class SocietyUser : IObjectWithState, IIdentifiableModel<Guid>
    {
        public SocietyUser()
        {

        }
        public Guid Id { get; set; }
        public Guid SocietyId { get; set; }

        public Guid BuildingId { get; set; }
        public Guid FlatId { get; set; }
        public Guid UserId { get; set; }

        public Guid? RoleId { get; set; }
        public Guid? ApplicationId { get; set; }

        public ObjectState ObjectState { get; set; }
    }
}

File: Apollo.Web/ClientApp/src/app/backoffice/viewmodel/user-mgmt-vm/societyuser.ts
import { IObjectWithState } from 'src/app/common/iobjectwithstate';
import { ObjectState } from 'src/app/common/enums';

export class SocietyUser implements IObjectWithState {
  objectState: ObjectState;
  societyId: string;
  buildingId: string;
  flatId: string;
  roleId: string;
}
